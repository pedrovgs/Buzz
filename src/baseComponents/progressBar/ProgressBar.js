import React from "react";
import { CircularProgress } from "material-ui";
import { isRunningTests } from "../../testUtils/utils";

export default class ProgressBar extends React.Component {
  render() {
    const mode = isRunningTests() ? "determinate" : "indeterminate";
    return (
      <CircularProgress
        id="progress"
        className="center"
        mode={mode}
        size={100}
        value={55}
      />
    );
  }
}
