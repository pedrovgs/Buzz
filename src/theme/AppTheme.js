import getMuiTheme from "material-ui/styles/getMuiTheme";

const AppTheme = getMuiTheme({
  fontFamily: "Roboto, sans-serif",
  palette: {
    primary1Color: "#f2db2c",
    primary2Color: "#33658a",
    accent1Color: "#bbaa00",
    accent2Color: "#003b5d",
    textColor: "#000000",
    alternateTextColor: "#ffffff"
  }
});

export default AppTheme;
