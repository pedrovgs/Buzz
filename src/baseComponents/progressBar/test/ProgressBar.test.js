jest.mock("../../../testUtils/utils");
const testUtilsMock = require("../../../testUtils/utils");
import React from "react";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import AppTheme from "../../../theme/AppTheme";
import Enzyme from "enzyme";
import { createSerializer } from "enzyme-to-json";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import ProgressBar from "../ProgressBar";

expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));
Enzyme.configure({ adapter: new Adapter() });

describe("ProgressBar", () => {
  it("shows the application progress bar as determinated if we are running tests", () => {
    givenTheAppIsRunningTests();

    const progressBar = renderProgressBar();

    expect(progressBar).toMatchSnapshot();
  });

  it("shows the application progress bar as indeterminated if we are not running tests", () => {
    givenTheAppIsNotRunningTests();

    const progressBar = renderProgressBar();

    expect(progressBar).toMatchSnapshot();
  });

  function renderProgressBar() {
    return toJson(
      Enzyme.render(
        <MuiThemeProvider muiTheme={AppTheme}>
          <ProgressBar />
        </MuiThemeProvider>
      )
    );
  }

  function givenTheAppIsNotRunningTests() {
    testUtilsMock.default.__configureEnvironmentAsRunningTests(false);
  }

  function givenTheAppIsRunningTests() {
    testUtilsMock.default.__configureEnvironmentAsRunningTests(true);
  }
});
