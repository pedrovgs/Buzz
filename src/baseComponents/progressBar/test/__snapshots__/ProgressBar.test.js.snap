// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar shows the application progress bar as determinated if we are running tests 1`] = `
<div
  id="progress"
  max="100"
  min="0"
  mode="determinate"
  style="position:relative;display:inline-block;width:100px;height:100px;mui-prepared:"
  value="55"
>
  <div
    style="width:100px;height:100px;display:inline-block;transition:20s transform linear 0ms;transition-timing-function:linear;mui-prepared:"
  >
    <svg
      style="width:100px;height:100px;position:relative;mui-prepared:"
      viewBox="0 0 100 100"
    >
      <circle
        cx="50"
        cy="50"
        fill="none"
        r="48.25"
        stroke-miterlimit="20"
        stroke-width="3.5"
        style="stroke:#f2db2c;stroke-linecap:round;transition:0.3s all linear 0ms;stroke-dasharray:166.7400300892783, 303.16369107141503;mui-prepared:"
      >
        
      </circle>
    </svg>
  </div>
</div>
`;

exports[`ProgressBar shows the application progress bar as indeterminated if we are not running tests 1`] = `
<div
  id="progress"
  max="100"
  min="0"
  mode="indeterminate"
  style="position:relative;display:inline-block;width:100px;height:100px;mui-prepared:"
  value="55"
>
  <div
    style="width:100px;height:100px;display:inline-block;transition:20s transform linear 0ms;transition-timing-function:linear;mui-prepared:"
  >
    <svg
      style="width:100px;height:100px;position:relative;mui-prepared:"
      viewBox="0 0 100 100"
    >
      <circle
        cx="50"
        cy="50"
        fill="none"
        r="48.25"
        stroke-miterlimit="20"
        stroke-width="3.5"
        style="stroke:#f2db2c;stroke-linecap:round;transition:1.5s all ease-in-out 0ms;mui-prepared:"
      >
        
      </circle>
    </svg>
  </div>
</div>
`;
